java/src/main/java/io/cucumber/java/JavaDefaultDataTableCapitalCaseTransformerDefinition.java

```diff
@@ -0,0 +1,77 @@
+package io.cucumber.java;
+
+import io.cucumber.core.backend.Lookup;
+import io.cucumber.core.runtime.Invoker;
+import io.cucumber.datatable.TableCellByTypeTransformer;
+import io.cucumber.datatable.TableEntryByTypeTransformer;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class JavaDefaultDataTableCapitalCaseTransformerDefinition extends JavaDefaultDataTableEntryTransformerDefinition {
+    JavaDefaultDataTableCapitalCaseTransformerDefinition(Method method, Lookup lookup) {
+        super(method, lookup);
+    }
+
+    @Override
+    public TableEntryByTypeTransformer tableEntryByTypeTransformer() {
+        return super.tableEntryByTypeTransformer();
+    }
+
+    @Override
+    Object execute(Map<String, String> fromValue, Type toValueType, TableCellByTypeTransformer cellTransformer) throws Throwable {
+        Object[] args;
+        if (method.getParameterTypes().length == 3) {
+            args = new Object[]{fromValue, toValueType, cellTransformer};
+        } else {
+            args = new Object[]{fromValue, toValueType};
+        }
+        args[0] = convertToCamelCase((Map<String, Object>)args[0]);
+        return Invoker.invoke(lookup.getInstance(method.getDeclaringClass()), method, args);
+    }
+
+    private Map<String, Object> convertToCamelCase(Map<String, Object> map) {
+        Map<String, Object> newMap = new HashMap<>();
+        CamelCaseStringConverter converter = new CamelCaseStringConverter();
+        for (Map.Entry<String, Object> entry : map.entrySet()) {
+            newMap.put(converter.map(entry.getKey()), entry.getValue());
```

Comment:
What will happen when I have a table that looks like this?

```
| Example Header | example header | example Header |
|  2             |  3             |  4             | 
```

Reply:
Given the complexity of this function and given that it doesn't have any tests yet and given that in its current state is rather hard to test you may want to consider reorganizing this method and and the `CamelCaseStringConverter` so you have a single test able unit.